core/deployment/src/main/java/io/quarkus/deployment/dev/CompilationProvider.java

```diff
@@ -39,6 +39,7 @@ class Context {
         private final File outputDirectory;
         private final Charset sourceEncoding;
         private final List<String> compilerOptions;
+        private final List<String> kotlinCompilerOptions;
```

Comment:
Is this really necessary? There will be a language-specific impl of CompilationProvider, e.g. JavaCompilationProvider, KotlinCompilationProvider. Ideally, compilationOptions should be relevant to the impl.

Reply:
Right, we could introduce a Map with all provider specific impl and compiler would consume its own options.

Reply:
well in fact we can directly deal with it without a map, I will push a fix.

Reply:
I pushed a fix, context are based on the source path, If the source path contains `kotlin`, I use the `kotlinCompilerOptions`, otherwise I use `compilerOptions`

Reply:
Sorry @glefloch I should have payed more attention. Looking at it closer, the same context can be re-used for multiple compilers. So we do need to be able to provide different compiler options from the same context. So yes, it either has to be map-based or what you had originally. A Map-based approach seems to be more generic (will allow to support Scala compiler options, for example, too) but will also require introducing a unique compiler provider key.